alf/utils/dist_utils.py

```diff
@@ -412,6 +424,19 @@ def concentration0(self):
     def concentration1(self):
         return self._concentration1
 
+    def rsample(self, sample_shape=()):
+        """We override the original ``rsample()`` in order to clamp the output
+        to avoid `NaN` and `Inf` values in the gradients. See Pyro's
+        ``rsample()`` implementation in
+        `<https://docs.pyro.ai/en/dev/_modules/pyro/distributions/affine_beta.html#AffineBeta>`_.
+        """
+        x = super(Beta, self).rsample(sample_shape)
+        if self._eps is None:
+            eps = torch.finfo(x.dtype).eps
+        else:
+            eps = self._eps
```

Comment:
move 434-437 to constructor

Reply:
> move 434-437 to constructor

I thought of doing this, but there is no way to get x.dtype in the constructor, unless the init accepts a dtype in advance.

Reply:
It's self._dirichlet.concentration.dtype